charts/metrics-server/templates/deployment.yaml

```diff
@@ -33,7 +33,9 @@ spec:
         {{- toYaml . | nindent 8 }}
       {{- end }}
     spec:
-      schedulerName: {{ .Values.schedulerName }}
+      {{- with .Values.schedulerName }}
+      schedulerName: {{ . | quote }}
```

Comment:
I still don't think `quote` is necessary here, if you look at the chart as a whole we don't add this layer of protection. The general rule is that if a valid value may be interpreted as another type use `quote` otherwise 50%+ of the chart characters would be `quote`. If you want to protect against the wrong type being used just add the field to the JSON schema so the Helm wont even attempt to render with the wrong type.

Reply:
TBH I'm a bit surprised, as explicit quoting seems to be quite standard in the Helm ecosystem.

It is recommended in the official [Best Practices](https://helm.sh/docs/chart_best_practices/values/#make-types-clear) as well as the [Chart Development guideline](https://helm.sh/docs/howto/charts_tips_and_tricks/#quote-strings-dont-quote-integers).

Indeed I see few other examples, maybe not 50% of the chart though.

For example `1234` would be a totally valid name for a Kubernetes namespace, however with current configuration in [service.yaml#L5](https://github.com/kubernetes-sigs/metrics-server/blob/master/charts/metrics-server/templates/service.yaml#L5) the chart would render the namespace value as an integer (`namespace: 1234` ) which is invalid syntax.

```
$ helm -n 1234 template ./charts/metrics-server

---
# Source: metrics-server/templates/service.yaml
apiVersion: v1
kind: Service
metadata:
  name: release-name-metrics-server
  namespace: 1234
[...]

$ kubectl create ns 1234
namespace/1234 created

$ helm -n "1234" install test metrics-server/metrics-server
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: unable to decode "": json: cannot unmarshal number into Go struct field ObjectMeta.metadata.namespace of type string
```

If you have a look at bitnami's charts (ie. [grafana](https://github.com/bitnami/charts/blob/main/bitnami/grafana/templates/configmap.yaml#L10)) they add explicit quoting for that very reason.

```
$ helm -n 1234 template oci://registry-1.docker.io/bitnamicharts/grafana

Pulled: registry-1.docker.io/bitnamicharts/grafana:10.0.9
Digest: sha256:1fd037e9e038230968457f85fb0f19e48c83e0c0f84f8b1fde6398f9372dd689
---
# Source: grafana/templates/networkpolicy.yaml
kind: NetworkPolicy
apiVersion: networking.k8s.io/v1
metadata:
  name: release-name-grafana
  namespace: "1234"
[...]
```

For the exact same reason, `--set schedulerName="1234"` would fail where `1234` is a totally valid name for a scheduler. You would have to explicitly use `--set-string` then.


That is not a big issue though, I just wanted  to share my surprise and maybe get your point of view regarding this. Could be the topic of a future PR.

Reply:
Note I did push without `quote` though

Reply:
If you want to define chart value types that's what the schema is for, adding extra `quote` calls adds template complexity and makes the resulting output more complex. So for your example you can use the schema to make sure that `"1234"` is used instead of `1234`.